packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -506,7 +569,8 @@ export function makeVatKeeper(
         const slot = k.slice(prefix.length);
         if (!slot.startsWith('k')) {
           const vatSlot = slot;
-          const kernelSlot = kvStore.get(k);
+          const kernelSlot =
+            kvStore.get(k) || assert.fail('getKeys ensures get');
```

Comment:
this can be a `getRequired`, yeah?

Reply:
`getRequired` felt like a bit of a performance hit. I was trying to just satisfy the typechecker.

I suppose the `||` is a runtime check too, though. Hm.

Reply:
The thought was: `getRequired()` is used where the "schema" says the key can't be missing, but it's not statically evident from the code.

Where it's statically evident that `.get(k)` won't return `undefined` because we got `k` from `getKeys` or just checked it with `has`, the `|| assert.fail()` idiom tells the typechecker.

I'm not a fan of the `has()` followed by `get()` idiom. Feels fragile and requires 2 round trips to the DB.

And I'd like to see `getEntries()` replace `getKeys()` followed by `get()`. But that felt out of scope.

Reply:
Ok, I just figured that since the implementation of `getRequired` is identical to `get || assert`, then using `getRequired` would make the code shorter and easier to read. My habit has been that `get` means I'm prepared to tolerate it being missing, and `getRequired` means I'm not (DB state is confused and it's best to halt right away). But I'm happy to see it written either way.